src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -1969,7 +1969,13 @@ private static void AddTraversalReferencesTarget(ProjectInstance traversalProjec
                 outputItemAsItem = "@(" + outputItem + ")";
             }
 
-            ProjectTargetInstance target = traversalProject.AddTarget(targetName ?? "Build", String.Empty, String.Empty, outputItemAsItem, null, String.Empty, String.Empty, String.Empty, String.Empty, false /* legacy target returns behaviour */);
+            string correctedTargetName = targetName ?? "Build";
+
+            if (traversalProject.Targets.Select(target => target.Key).Contains(correctedTargetName))
+            {
+                traversalProject.RemoveTarget(correctedTargetName);
+            }
```

Comment:
You can just do this unconditionally; `RemoveTarget` doesn't throw if it's not there.

Reply:
But this depends on the "how did pre-16.4 behave" question WRT having stub targets in before/after.

Reply:
Sorry—I should have mentioned this rather than just pushing the change. In 16.3, if you have a before...targets or an after...targets with a Build target, the Build target is overwritten by the generated target.